### ISSUE TITLE
Filters and sorting does not work with server driven datagrid and pagination

### ISSUE THREAD

**Issue by nohanna** [vmware/clarity#3112]
## Describe the bug

When using server driven pagination: every page is loading 10 items by a request on my server. 
I noticed that filters and sorting were not working anymore due to *ngFor instead of *clrDgItems.

I also tried with *clrDgItems (I know that makes no sense but I tried anyways), filters and sorting were working with pagination but this was not updating my model when I changed the page.

I found a hack using *clrDgItems that updates my model (see additional notes below).

This might be a duplicate of https://github.com/vmware/clarity/issues/887 but there was no "explicit" code or explanation.

## How to reproduce

```
<clr-datagrid (clrDgRefresh)="refresh($event)" [(clrDgSelected)]="selected">
    <clr-dg-column [clrDgField]="'name'">Name</clr-dg-column>
    <clr-dg-column [clrDgField]="'age'">Age</clr-dg-column>
     
    <clr-dg-row *ngFor="let user of (users$ | async)" [clrDgItem]="user">
        <clr-dg-cell>{{ user.name }}</clr-dg-cell>
        <clr-dg-cell>{{ user.age }}</clr-dg-cell>
    </clr-dg-row>

    <clr-dg-footer>
        {{ pagination.firstItem + 1 }} - {{ pagination.lastItem + 1 }} of {{ totalElements$ | async }} users
        <clr-dg-pagination #pagination [(clrDgPage)]="page" [clrDgLastPage]="totalPages$ | async"> 
        </clr-dg-pagination>
    </clr-dg-footer>
</clr-datagrid>
```

Use a server driven datagrid with *ngFor and a pagination in the datagrid.

## Expected behavior

Filters and sorting should work with server driven datagrid and pagination and even with server driven pagination.

## Versions

**App**

* Angular: 7.2.4
* Clarity: 0.13.10 (tested with 1.0.5 too)

**Device:**

* OS: CentOS 7
* Browser: all browsers

## Additional notes

I found a hack to get this working by adding another datagrid below the first one with pagination that refreshes the first one and put some css tricks to hide the second datagrid.

Note that I used *clrDgItems instead of *ngFor to get this work.
```
<clr-datagrid [(clrDgSelected)]="selected">
    <clr-dg-column [clrDgField]="'name'">Name</clr-dg-column>
    <clr-dg-column [clrDgField]="'age'">Age</clr-dg-column>
     
    <clr-dg-row *clrDgItems="let user of (users$ | async)" [clrDgItem]="user">
        <clr-dg-cell>{{ user.name }}</clr-dg-cell>
        <clr-dg-cell>{{ user.age }}</clr-dg-cell>
    </clr-dg-row>
</clr-datagrid>
<clr-datagrid (clrDgRefresh)="refresh($event)">
    <clr-dg-footer>
        {{ pagination.firstItem + 1 }} - {{ pagination.lastItem + 1 }} of {{ totalElements$ | async }} users
        <clr-dg-pagination #pagination [(clrDgPage)]="page" [clrDgLastPage]="totalPages$ | async"> 
        </clr-dg-pagination>
    </clr-dg-footer>
<clr-datagrid>
```

**Comment by youdz** [comment]
You are getting paginated results from your server (10 items per page), but you want the filtering and sorting to happen on the client side?
How would that work? When I display the 10 items on the first page of a datagrid that sorts by "name", I expect these 10 to be the first names in alphabetical order **across the entire dataset**. Just sorting whatever happens to be on the first page returned by the server doesn't accomplish anything. And the same applies to filters.

In other words, if the server handles pagination then it clearly needs to also handle sorting and pagination. Otherwise the "first page" concept is meaningless.

**Comment by nohanna** [comment]
Filtering on the client side is only a pipe. It's working as expected with my hack, it's only filtering on the current page, and when you change the page, it calls clrDgRefresh and reapply the filter on the new page.
There is no need to filter on server side if I don't need it for some reasons.

I just was wondering why wouldn't it work if you just want to use a "pipe" filter on client side instead of implementing a server side filter. 
Creating another datagrid to achieve this is, for sure, not the right way.

**Comment by youdz** [comment]
You might see it as a pipe, but the resulting UX is extremely confusing. Our designs are thought out and researched in order to provide components that are intuitive, consistent and instantly usable by the end-user. 

In your case, if I have a datagrid with 10 pages and am looking for an item with the name “John”, I will try filtering by name. If the user in question in on the 4th page, your version would show me 0 results and I would assume there is no “John” in the dataset. I’m not saying it’s not technically possible, I’m saying it is contrary to our goal of high-quality user experience, so we recommend heavily against it and will not support it as an option. 

I hope this helps understand d why I closed the issue.

**Comment by nohanna** [comment]
Yes, I understand this point. 
Well in my case, a basic filter was fine. But it seems not to fit really well with my pagination.

Anyways, I will create a custom filter for this case.
Thanks for your answer.
